src/Action/SetObjectFieldValueAction.php

```diff
@@ -164,4 +161,37 @@ public function __invoke(Request $request): JsonResponse
 
         return new JsonResponse($content, Response::HTTP_OK);
     }
+
+    private function getFieldDataTransformer(
```

Comment:
I would prefer a more generic way like a `DataTransformerResolver`. This way we can add new transformers more easily.

Reply:
Do you suggest just putting this method in a separate class?

Reply:
Like this?

```php
final class DataTransformerResolver
{
    private $modelManager;

    public function __construct($modelManager)
    {
        $this->modelManager = $modelManager;
    }

    private function resolve(FieldDescriptionInterface $fieldDescription): ?DataTransformerInterface
    {
        $dataTransformer = $fieldDescription->getOption('data_transformer');

        // allow override transformers for 'date', 'boolean' and 'choice' field types
        if ($dataTransformer instanceof DataTransformerInterface) {
            return $dataTransformer;
        }

        // Handle date type has setter expect a DateTime object
        if ('date' === $fieldDescription->getType()) {
            return new DateTimeToStringTransformer();
        }

        // Handle boolean type transforming the value into a boolean
        if ('boolean' === $fieldDescription->getType()) {
            return new BooleanToStringTransformer('1');
        }

        // Handle entity choice association type, transforming the value into entity
        if ('choice' === $fieldDescription->getType()) {
            $className = $fieldDescription->getOption('class');

            if (null !== $className && $className === $fieldDescription->getTargetEntity()) {
                return new ModelToIdTransformer($this->modelManager, $className);
            }
        }

        return null;
    }
}
```

Reply:
Or maybe you are interested in the option with individual services that are binded into a resolver?

Reply:
Or like this?
```php
final class ResolverFieldDataTransformer implements FieldDataTransformer
{
    private $dataTransformers;

    public function __construct(array $dataTransformers)
    {
        $this->dataTransformers = $dataTransformers;
    }

    public function transform($value, FieldDescriptionInterface $fieldDescription)
    {
        $dataTransformer = $fieldDescription->getOption('data_transformer');

        // allow override predefined transformers
        if ($dataTransformer instanceof DataTransformerInterface) {
            return $dataTransformer->reverseTransform($value);
        }

        $type = $fieldDescription->getType();

        if (array_key_exists($type, $this->dataTransformers)) {
            return $this->dataTransformers[$type]->transform($value, $fieldDescription);
        }

        return $value;
    }
}

final class ChoiceFieldDataTransformer implements FieldDataTransformer
{
    private $modelManager;

    public function __construct($modelManager)
    {
        $this->modelManager = $modelManager;
    }

    public function transform($value, FieldDescriptionInterface $fieldDescription)
    {
        $className = $fieldDescription->getOption('class');

        if (null !== $className && $className === $fieldDescription->getTargetEntity()) {
            return (new ModelToIdTransformer($this->modelManager, $className))->reverseTransform($value);
        }

        return $value;
    }
}

$resolver = new ResolverFieldDataTransformer([
    'choice' => new ChoiceFieldDataTransformer($modelManager),
]);
```